api/src/OpenTelemetry/Attributes.hs

```diff
@@ -75,7 +74,10 @@ data Attributes = Attributes
   , attributesCount :: {-# UNPACK #-} !Int
   , attributesDropped :: {-# UNPACK #-} !Int
   }
-  deriving stock (Show, Eq)
+  deriving stock (Show, Generic, Eq, Ord)
+
+
+instance Hashable Attributes
```

Comment:
Needed for `instance Hashable InstrumentationLibrary`

Reply:
I can't think of a reason at this point where `InstrumentationLibrary` or this would benefit from supplying a `Hashable` instance.

Reply:
It looks like one of the Processors uses the Hashable instance of InstrumentationLibrary
<img width="913" alt="Screenshot 2024-06-10 at 1 45 56 PM" src="https://github.com/iand675/hs-opentelemetry/assets/169551307/f0711a5e-10e2-4798-9b1e-0f5b779f558f">

Reply:
Ah I see. The in the previous version, it was a name + version pair, which makes sense as a thing that's hashable. By adding attributes as well, you could technically end up with two `InstrumentationLibrary` values with the same name and version, but different attributes, which is a little bit odd here 🤔 . I think it would make sense to have the `InstrumentationLibrary` keep the shape prior to this PR, but have a `HashMap InstrumentationLibrary ...attributes...` that we use to add on attributes upon export.

Reply:
[From the docs](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#get-a-logger):
<img width="1292" alt="Screenshot 2024-06-10 at 4 37 16 PM" src="https://github.com/iand675/hs-opentelemetry/assets/169551307/66566aa7-978f-4bc0-9cf4-91afbf4c8775">

Reply:
It still might be worth making it impossible to make two loggers with different attributes but the same identity, but the docs says that is a user error so I'm not sure if we need to worry about that.